static/js/giphy.js

```diff
@@ -0,0 +1,81 @@
+import {renderGrid} from "@giphy/js-components";
+import {GiphyFetch} from "@giphy/js-fetch-api";
+import $ from "jquery";
+import {throttle} from "throttle-debounce";
+
+import render_giphy_picker from "../templates/giphy_picker.hbs";
+
+import * as compose_ui from "./compose_ui";
+
+const gf = new GiphyFetch(page_params.giphy_api_key);
+let search_term = "";
+
+function fetchGifs(offset) {
+    const config = {
+        offset,
+        limit: 25,
+    };
+    if (search_term === "") {
+        // Get the trending gifs by default.
+        return gf.trending(config);
+    }
+    return gf.search(search_term, config);
+}
+
+export function renderGIPHYGrid(targetEl) {
+    const render = () =>
+        // See https://github.com/Giphy/giphy-js/blob/master/packages/components/README.md#grid
+        // for detailed documentation.
+        renderGrid(
+            {
+                width: 300,
+                fetchGifs,
+                columns: 3,
+                gutter: 6,
+                noLink: true,
+                // Hide the user attribution that appears over a GIF
+                hideAttribution: true,
+                onGifClick: (props) => {
+                    $("#compose_box_giphy_grid").popover("hide");
+                    // TODO: Decide whether to include the search term here or not based on
+                    // feedback.
```

Comment:
I think we're unlikely to want to include the search term here -- that doesn't seem to me like part of the expected interface.

Reply:
Okay.